### PR TITLE
Add gradle target and github workflow for cross-langauge yaml tests.

### PR DESCRIPTION
**Pull request by robertwb**
This should have caught the issue at https://github.com/apache/beam/issues/30776

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

File: .github/workflows/beam_PreCommit_Yaml_Xlang_Direct.yml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+      - name: Setup repository
+        uses: ./.github/actions/setup-action
+        with:
+          comment_phrase: ${{ matrix.job_phrase }}
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          github_job: ${{ matrix.job_name }} (${{ matrix.job_phrase }})
+      - name: Setup environment
+        uses: ./.github/actions/setup-environment-action
+        with:
+          python-version: |
```
Comment (damccorm) [inline .github/workflows/beam_PreCommit_Yaml_Xlang_Direct.yml:82]: Do we actually need 2 python versions?

Comment (damccorm) [reply .github/workflows/beam_PreCommit_Yaml_Xlang_Direct.yml:82]: I think we're just running against 3.11 here

File: .github/workflows/beam_PreCommit_Yaml_Xlang_Direct.yml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+on:
+  pull_request_target:
+    paths: ['release/trigger_all_tests.json', 'model/**', 'sdks/python/**']
+  issue_comment:
+    types: [created]
+  push:
+    tags: ['v*']
+    branches: ['master', 'release-*']
+    paths: [ "model/**","sdks/python/**","release/**",".github/workflows/beam_PreCommit_Yaml_Xlang_Direct.yml"]
```
Comment (damccorm) [inline .github/workflows/beam_PreCommit_Yaml_Xlang_Direct.yml:26]: Should this trigger on some java directories too? I think the java directory change is the one that actually caused the failure (and I expect this to generally be the case)

File: sdks/python/build.gradle [hunk 3]
```diff
@@ -107,6 +107,26 @@ tasks.register("generateYamlDocs") {
...
+  dependsOn ":sdks:java:io:expansion-service:build"
+  dependsOn ":sdks:java:io:google-cloud-platform:expansion-service:build"
+
+  doLast {
+    exec {
+      executable 'sh'
+      args '-c', "${envdir}/bin/pytest -v apache_beam/yaml/integration_tests.py"
+    }
+  }
+  outputs.file "${buildDir}/yaml-ref.html"
```
Comment (damccorm) [inline sdks/python/build.gradle:127]: Is this output correct?

File: .github/workflows/beam_PreCommit_Yaml_Xlang_Direct.yml (hunk 1 above)
Comment (robertwb) [reply .github/workflows/beam_PreCommit_Yaml_Xlang_Direct.yml:82]: It looks like the gradle commands are using 3.8 by default. We probably don't need both.

File: sdks/python/build.gradle (hunk 3 above)
Comment (robertwb) [reply sdks/python/build.gradle:127]: Removed.

File: .github/workflows/beam_PreCommit_Yaml_Xlang_Direct.yml (hunk 2 above)
Comment (robertwb) [reply .github/workflows/beam_PreCommit_Yaml_Xlang_Direct.yml]: Yeah. It's hard to know where to draw the line. I've added some candidates, and this should get run on postcommit/release as well.

**Review by damccorm — APPROVED** [review]
Thanks

**Pull request merged**